### DISCUSSION TITLE
v3.6.X

### DISCUSSION THREAD

**Discussion by Tanjitsu** [scripthookvdotnet/scripthookvdotnet#1405] [General]
It has been over a year since SHVDN updated and there has been changes to the game in that time.

I would suggest it would be worth releasing a v3.6.x release containing the current v3.6.0 branch but with updated enum hash values prior to you releasing the mass changes on the master branch.

The previous re-write from v2 to v3 ended up with both api's being accessible.
This allowed older mods who are not being maintained to still continue to work if functionality was not broken by the game.

The large amount of changes currently made to the master branch will mean old v3 mods will not work on the next release unless the Dev returns and re-writes the mod.

This leaves the users with 2 choices.
1. use the new v4.0 but can only use very new mods.
2. use the old version v.3.6 but are missing over a years worth of vehicle definitions and ped hashes ect.

There are methods to collect and call the hashes direct from the game irrelevant of the SHVDN definitions but the point of the definitions is to make life simpler for the modder... isn't it?

**Comment by kagikn** [comment]
> The large amount of changes currently made to the master branch will mean old v3 mods will not work on the next release unless the Dev returns and re-writes the mod.

Could you tell me a mod that will not work at the commit 36a892cb881be67f091d52713854d3224e5c4f1e, which is the latest one on the main branch as of writing? I don't mean to break any compatibilities for existing mods in minor updates. Some script mods that uses some audio libraries may not work at that commit iirc, but if that is because we have the main thread sleep when our logic is performing, we can't make them work until we find a way to hook some functions in the exe to do our tick operations. We have to hook some of them before we won't have RPH crash by disabling try-carching any exceptions (by running .NET code on a fiber and fucking up the Vectored Exception Handler ultimately) and we can run our logic on the main thread without fiber stuff (the latter is not achieved atm).

**Reply by Hydraxonn** [reply]
This is actually incredibly useful information. I was getting crashes with the latest SHVDN commit, and NAudio was the cause.

**Reply by Hydraxonn** [reply]
Spoke too soon. A couple days later, no changes to my game folder and the random/startup crashes caused by SHVDN are back. I can also begin taking apart my scripts folder to see if its a specific mod causing the issues. I suppose this won't truly be solved until SHVDN logic runs on the main thread without using fibers.

**Comment by Tanjitsu** [comment]
Hey,

> Could you tell me a mod that will not work at the commit

Not specifically, I will check a few to confirm.

I would assume its less by reference and more by functionality that the underlying changes have made.
